### ISSUE TITLE
[BUG] FileServer dont work with Gzip&"304 Not Modified"

### ISSUE THREAD

**Issue by huiminghao** [kataras/iris#1561]
**Describe the bug**
A demo drived from fileserver/basic example.  
When `Gzip` set to `false`, the http server will response 304 when not modified.
While `Gzip` set to `true`, the http server will always response gziped body, ignored the 304 situation.

**To Reproduce**
```
	ws.app = iris.New()
	ws.app.HandleDir("/", rootdir, iris.DirOptions{
		IndexName: "/index.html",
		Gzip:      false,
		ShowList:  false,
	})
	go func() {
		err := ws.app.Run(iris.Addr("0.0.0.0:8080"), iris.WithoutInterruptHandler)
		if err != nil {
			ws.errros <- err
		}
	}()
```
```
	ws.app = iris.New()
	ws.app.HandleDir("/", rootdir, iris.DirOptions{
		IndexName: "/index.html",
		Gzip:      true,
		ShowList:  false,
	})
	go func() {
		err := ws.app.Run(iris.Addr("0.0.0.0:8080"), iris.WithoutInterruptHandler)
		if err != nil {
			ws.errros <- err
		}
	}()
```

**Expected behavior**
Gzip should support 304.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Debian 10
 - iris version: v12.1.8
 - go version: 1.14.2 amd64

**Comment by kataras** [comment]
Yes, this should be fixed on master branch, future v12.2.0, please test it through `go get -u github.com/kataras/iris@master` and write down

**Comment by kataras** [comment]
Yes, this should be fixed on master branch some days ago, future v12.2.0, please test it through `go get -u github.com/kataras/iris/v12@master` and write down if you still have issues @huiminghao. Look #1554 too.
